Automattic/wp-calypso
# Card: Style cursor as pointer when onClick prop exists

Rationale: Previously, some instances of `Card` (or components that internally render  `Card`s) used to include a workaround such as `href=„#“` (alongside `event.preventDefault()` in the `onClick` handler) in order to get a `pointer` style cursor.

Note that `Card` is one of our most pervasive components. However, I think this PR makes a reasonable assumption and isn't prone to break existing functionality.

To test:
Apply this patch:

```diff
diff --git a/client/my-sites/site-settings/manage-connection/disconnect-site-link.jsx b/client/my-sites/site-settings/manage-connection/disconnect-site-link.jsx
index f195984..9140bfb 100644
--- a/client/my-sites/site-settings/manage-connection/disconnect-site-link.jsx
+++ b/client/my-sites/site-settings/manage-connection/disconnect-site-link.jsx
@@ -23,9 +23,7 @@ class DisconnectSiteLink extends Component {
                dialogVisible: false,
        };
 
-       handleClick = event => {
-               event.preventDefault();
-
+       handleClick = () => {
                this.setState( {
                        dialogVisible: true,
                } );
@@ -51,7 +49,6 @@ class DisconnectSiteLink extends Component {
                                <QuerySitePlans siteId={ siteId } />
 
                                <SiteToolsLink
-                                       href="#"
                                        onClick={ this.handleClick }
                                        title={ translate( 'Disconnect from WordPress.com' ) }
                                        description={ translate(
```

Go to `/settings/manage-connection/:site` (with a JP site slug). 

Hover over the 'Disconnect from WordPress.com' button at the bottom of the page, verify that the cursor changes to a pointer.

![image](https://user-images.githubusercontent.com/96308/31471251-3b2258a6-aee8-11e7-87e6-ece683dccf69.png)

Bonus points if you apply the above patch to `master` and verify that the cursor is a text cursor.

---

Motivated by https://github.com/Automattic/wp-calypso/pull/19270#discussion_r147683605
